tensorflow/python/framework/tensor_util.py

```diff
@@ -219,6 +219,18 @@ def _NotNone(v):
     return v
 
 
+def _FilterTuple(v):
+  if not isinstance(v, (list, tuple)):
+      return v
+  if isinstance(v, tuple):
+      if not True in [isinstance(x, (list, tuple)) for x in v]:
```

Comment:
this creates a python list each time. use the function `any(isinstance(x, (list, tuple)) for x in v)`